include/zephyr/bluetooth/mesh/access.h

```diff
@@ -1118,6 +1118,24 @@ bool bt_mesh_model_is_extended(struct bt_mesh_model *model);
  */
 int bt_mesh_comp_change_prepare(void);
 
+/** @typedef bt_mesh_comp2_cb_t
+ *  @brief Composition data page 2 callback.
+ *
+ *  @param buf Composition data page 2 buffer to fill.
+ *
+ *  @return 0 on success or negative return code.
+ */
+typedef int (*bt_mesh_comp2_cb_t)(struct net_buf_simple *buf);
+
+/** @brief Register a callback providing composition data page 2 of the device.
+ *
+ *  Register a callback that provides Network Lighting Control (NLC)
+ *  information for composition data page 2 of the device.
+ *
+ *  @param cb Callback structure.
+ */
+void bt_mesh_comp_data_page_2_cb_reg(bt_mesh_comp2_cb_t cb);
```

Comment:
I like this 👍 I will do another iteration on implementation.

Reply:
It is very unlikely that preprocessor magic will work. Registering pointer has a small risk that pointer needs to be filled in appropriately or valid, may be this is not a concern since NCS implementation that calls this API will make sure of that.

Reply:
I think it is the same risk level as calling any of our API with NULL pointer, or calling API from a preemptable thread. This is how we instantiate CDP0 and should work well for CDP2. Perhaps this is the solution to avoid storing CDP1 in RAM as we do now: https://github.com/zephyrproject-rtos/zephyr/blob/a15e7d2326d9e8896401be479fae56f30dde71bb/subsys/bluetooth/mesh/access.c#L82?